src/Compilers/CSharp/Test/Syntax/Parsing/ParserErrorMessageTests.cs

```diff
@@ -2674,15 +2706,186 @@ public static class GenExtensions<X>
-Diagnostic(ErrorCode.ERR_BadOutWithThis, "out"),
-Diagnostic(ErrorCode.ERR_BadOutWithThis, "out"),
-Diagnostic(ErrorCode.ERR_BadOutWithThis, "out"));
+                // (6,33): error CS81250:  The parameter modifier 'out' cannot be used with 'this' 
+                //     public static void Foo(this out int i) {}
+                Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "this").WithLocation(6, 33),
+                // (8,36): error CS81250:  The parameter modifier 'out' cannot be used with 'this' 
+                //     public static void Foo<T>(this out T t) {}
+                Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "this").WithLocation(8, 36),
+                // (10,40): error CS81250:  The parameter modifier 'out' cannot be used with 'this' 
+                //     public static void Foo<T,U,V>(this out U u) {}
+                Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "this").WithLocation(10, 40),
+                // (15,33): error CS81250:  The parameter modifier 'out' cannot be used with 'this' 
+                //     public static void Foo(this out int i) {}
+                Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "this").WithLocation(15, 33),
+                // (16,33): error CS81250:  The parameter modifier 'out' cannot be used with 'this' 
+                //     public static void Foo(this out X x) {}
+                Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "this").WithLocation(16, 33),
+                // (18,36): error CS81250:  The parameter modifier 'out' cannot be used with 'this' 
+                //     public static void Foo<T>(this out T t) {}
+                Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "this").WithLocation(18, 36),
+                // (19,36): error CS81250:  The parameter modifier 'out' cannot be used with 'this' 
+                //     public static void Foo<T>(this out X x) {}
+                Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "this").WithLocation(19, 36),
+                // (21,40): error CS81250:  The parameter modifier 'out' cannot be used with 'this' 
+                //     public static void Foo<T,U,V>(this out U u) {}
+                Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "this").WithLocation(21, 40),
+                // (22,40): error CS81250:  The parameter modifier 'out' cannot be used with 'this' 
+                //     public static void Foo<T,U,V>(this out X x) {}
+                Diagnostic(ErrorCode.ERR_BadParameterModifiers, "out").WithArguments("out", "this").WithLocation(22, 40));
+        }
+
+        [Fact]
+        public void BadInWithThis()
+        {
+            var test = @"
+public static class Extensions
+{
+    //No type parameters
+    public static void Test(this in int i) {}
```

Comment:
We want to eventually allow `this in` together with `this ref` as long as it is of a struct type.

It seems ok to disable that for now.